zone/client.cpp

```diff
@@ -6734,6 +6761,40 @@ bool Client::IsLootLockedOutOfNPC(uint32 npctype_id)
 	return false;
 };
 
+bool Client::IsLockedOutOfInstance(uint32 zone_id)
+{
+	if (zone_id == 0)
+		return false;
+
+	auto instancetItr = character_instance_lockouts.find(zone_id);
+
+	if (instancetItr != character_instance_lockouts.end())
+		return instancetItr->second.HasLockout(Timer::GetTimeSeconds());
+
+	return false;
+};
+
+uint32 Client::GetTargetZoneInstanceID(uint32 zone_id)
+{
+	if (zone_id == 0)
+		return GUILD_NONE;
+
+	auto instanceItr = character_instance_lockouts.find(zone_id);
+
+	if (instanceItr != character_instance_lockouts.end())
```

Comment:
Does this need a null check too?

Reply:
No. This is a reference and is stored by-ref, not by pointer.